### ISSUE TITLE
`state_dict()` for modules should be detached

### ISSUE THREAD

**Issue by dfalbel** [mlverse/torch#914]
Currently it looks like there's a bug that don't correctly detach the state dict.

https://github.com/mlverse/torch/blob/2b9ebe4e3693bf15f56851eecf54a0bd385d88eb/R/nn.R#L131-L133
